### DISCUSSION TITLE
Add a way to change name of Console and Remote Console `CommandSender`s

### DISCUSSION THREAD

**Discussion by molor** [PaperMC/Paper#10599] [Ideas]
### Is your feature request related to a problem?

Yes.

I want to change name of these senders to something that regular players cannot have (that doesn't match the regular expression `[a-zA-Z0-9_.]{2,16}` AKA valid player name).

Thus, in the case of public chat notifications about [remote] administration actions, and if a player "Rcon" or "CONSOLE" is connected to the server, it will be easy to understand who exactly performed some action (player CONSOLE or real console with name changed to, for example, "$erver" ($ isn't a valid symbol in player name) or "Remote Console" (player can't contain space in their name)).

### Describe the solution you'd like.

Add a setting to Paper configuration, i.e. `console-sender-name` and `remote-console-sender-name`, with `"default"` values by default to keep the current behavior.

### Describe alternatives you've considered.

Just check for the command sender class type in my plugins and use my custom `String`s instead of `CommandSender#getName()`. 

But, even if I omit the fact that I have a lot of plugins and support for "custom `String`s" needs to be added to each separately, then what to do with 3rd-party plugin notifications?...

### Other

PR should be very simple, so I hope someone can save me from having to fork the entire server just to change a couple of lines in its source...

Related files:
1. https://hub.spigotmc.org/stash/projects/SPIGOT/repos/craftbukkit/browse/src/main/java/org/bukkit/craftbukkit/command/CraftConsoleCommandSender.java#46
2. https://hub.spigotmc.org/stash/projects/SPIGOT/repos/craftbukkit/browse/src/main/java/org/bukkit/craftbukkit/command/CraftRemoteConsoleCommandSender.java#39

**Comment by lynxplay** [comment]
I wouldn't mind this, its not really a super common usecase, but not really harmful to add this.

**Comment by molor** [comment]
Another solution is to add a method like `setName(String/Component)` to these senders so their names can be changed via API to something different than current hard-coded default.

**Comment by Doc94** [comment]
hmm im curious about if is better a config or allow plugins alter this directly by code (method)

**Comment by molor** [comment]
Now I think that adding `setName(String)` + `name(Component)` and overriding `getName` + `name` of `ConsoleCommandSender` and `RemoteConsoleCommandSender`  would be better

**Comment by Machine-Maker** [comment]
I think a config for this is better. Plugins setting it seems to just be asking for conflicting plugins trying to set it to multiple things.

**Comment by molor** [comment]
Plugins can create `CommandSender`s via `Server#createCommandSender` and change its name, if necessary. In that case other plugins will not override their custom names. However this also requires addition of get/set methods to created `CommandSender`s too.
